### PR TITLE
fix: num_dir_sectors

### LINKED ISSUE #52: Corrupted file / Paths disappear when reading using ole32.dll after copy
I took the code from https://github.com/mdsteele/rust-cfb/issues/33#issuecomment-1475431426 to copy this cfb file:

https://github.com/mpcarr/aztec-quest/blob/fdf371ff7f792a46fedd987ef167491dde141a18/aztecquest-dev.vpx

When reading the copied file using windows `ole32.dll` most paths can no longer be found / read. The file size however seems to be about the same. Reading the file using `rust-cfb` still works.

This is the C# project that came up with below listing:
https://github.com/francisdb/ole32-cfb-list

```
================= Original ====================
Opening storage at c:\Users\franc\RustroverProjects\aztec-quest\aztecquest-dev.vpx
root\GameStg
root\GameStg\MAC
root\GameStg\Image0
root\GameStg\Image1
root\GameStg\Image2
root\GameStg\Image3
root\GameStg\Sound0
...
root\GameStg\Sound277
root\GameStg\GameItem0
...
root\GameStg\GameItem322
root\GameStg\Collection10
root\GameStg\Collection11
root\GameStg\Collection12
root\GameStg\Collection13
root\GameStg\Collection14
root\GameStg\Collection15
root\GameStg\Collection16
root\GameStg\Collection17
root\GameStg\Collection18
root\GameStg\CustomInfoTags
root\TableInfo
root\TableInfo\TableName
root\TableInfo\AuthorName
root\TableInfo\TableSaveRev
root\TableInfo\TableVersion
root\TableInfo\TableSaveDate
================= Copy ========================
Opening storage at c:\Users\franc\RustroverProjects\aztec-quest\aztecquest-copy.vpx
root\GameStg
root\GameStg\MAC
root\GameStg\Image0
root\GameStg\Image1
root\GameStg\Image2
root\GameStg\Image3
root\GameStg\Sound0
root\GameStg\Sound1
root\GameStg\Sound2
root\GameStg\Sound3
root\GameStg\Sound4
root\GameStg\Sound5
root\GameStg\Sound6
root\GameStg\Sound7
root\GameStg\Sound8
root\GameStg\Sound9
root\GameStg\Sound10
root\GameStg\Sound11
root\GameStg\Sound12
root\GameStg\Sound13
root\GameStg\Sound14
root\GameStg\Sound15
root\GameStg\Sound16
root\GameStg\Sound17
root\GameStg\Sound18
root\GameStg\Sound19
root\GameStg\Sound20
root\GameStg\Sound21
root\GameStg\Sound22
root\GameStg\Sound23
root\TableInfo
```

### Observations
* When on copy filtering out for example the Sound items the listing is again truncated to about the same length. So it's probably not a specific item that is causing further streams to be be found.
* When trying to read one of the unlisted streams I get this:
  `Exception has occurred: CLR/System.Runtime.InteropServices.COMException
An unhandled exception of type 'System.Runtime.InteropServices.COMException' occurred in ConsoleApp1.dll: 'The docfile has been corrupted. (0x80030109)'`
* wine ole32.dll has no issues reading the copied file
* even if the streams are very small this fails
* **only the first 32 items are found (including root/storages)**

### PR DESCRIPTION
**Pull request by francisdb**
Correctly update file header `num_dir_sectors` when new directory sectors are added.

fixes #52

### CODE REVIEW

File: src/internal/directory.rs [hunk 1]
```diff
@@ -413,13 +413,35 @@ impl<F: Write + Seek> Directory<F> {
...
         self.dir_entries.push(unallocated_dir_entry);
         Ok(stream_id)
     }
 
+    /// Increase header num_dir_sectors if version V4
+    /// note: not updating this value breaks ole32 compatibility
+    fn update_num_dir_sectors(&mut self) -> io::Result<()> {
+        let start_sector = self.dir_start_sector;
+        if self.version() == Version::V4 {
+            let num_dir_sectors = self.count_directory_sectors(start_sector)?;
```
Comment (francisdb) [inline src/internal/directory.rs:429]: I wanted to read the existing value and increase it but Directory has no `Read`.

Comment (francisdb) [reply src/internal/directory.rs:429]: On a second thought this would not fix existing files that are invalid. Current solution is ok.

**Comment by mdsteele** [comment]
Looks like the linter is unhappy, so the tests didn't run.  I just fixed the config (I think) so that next time the tests can still run even if the linter fails.

**Comment by francisdb** [comment]
@mdsteele applied cargo fmt

**Comment by francisdb** [comment]
Can I get a CI approval?

**Comment by mdsteele** [comment]
Thanks for the fix!

**Comment by francisdb** [comment]
Any plans for a release?

**Comment by mdsteele** [comment]
Published as [v0.10.0](https://crates.io/crates/cfb/0.10.0).

**Comment by francisdb** [comment]
> Published as [v0.10.0](https://crates.io/crates/cfb/0.10.0).

Thanks!

**Pull request merged**